benchmarks/kernels/benchmark_rmsnorm.py

```diff
@@ -30,7 +38,7 @@ def forward(
 
         variance = x.pow(2).mean(dim=-1, keepdim=True)
         x = x * torch.rsqrt(variance + self.variance_epsilon)
-        x = x.to(orig_dtype) * self.weight
+        x = x.to(orig_dtype) * (self.weight_bias + self.weight)
```

Comment:
Randomly stumbled upon this PR, and without a full in-depth review, I just wanted to mention that it is essential that weight_bias and weight are added in fp32. In certain gemma models, the normalization layers have weights that -- when added by the weight bias (1.0), produce incorrect results.

For example:
```py
>>> import torch
>>> weight = torch.tensor([15.8125], dtype=torch.bfloat16)
>>> weight + 1.0
tensor([16.7500], dtype=torch.bfloat16)
>>> weight.float() + 1.0
tensor([16.8125])
```

Not sure if this is considered here already, but just wanted to flag in case.


https://github.com/huggingface/transformers/blob/0fa49db1205e0a2745161ccac46184e7e46b6e2b/src/transformers/models/gemma/modular_gemma.py#L204